wsinfer/write_geojson.py

```diff
@@ -64,62 +66,48 @@ def _dataframe_to_geojson(df: pd.DataFrame, prob_cols: list[str]) -> dict:
-    df = pd.read_csv(input)
+def make_geojson(csv: Path, results_dir: Path) -> None:
+    filename = csv.stem
+    df = pd.read_csv(csv)
     prob_cols = [col for col in df.columns.tolist() if col.startswith("prob_")]
     if not prob_cols:
-        raise click.ClickException("Did not find any columns with prob_ prefix.")
+        raise KeyError("Did not find any columns with prob_ prefix.")
     geojson = _dataframe_to_geojson(df, prob_cols)
-    with open(output, "w") as f:
+    with open(results_dir / "model-outputs-geojson" / f"{filename}.json", "w") as f:
         json.dump(geojson, f)
 
 
-@click.command()
-@click.argument(
-    "results_dir",
-    type=click.Path(
-        exists=True, file_okay=False, dir_okay=True, path_type=Path, resolve_path=True
-    ),
-)
-@click.argument(
-    "output",
-    type=click.Path(exists=False, path_type=Path, resolve_path=True),
-)
-def togeojson(*, results_dir: Path, output: Path) -> None:
-    """Convert model outputs to GeoJSON format.
+def write_geojsons(csvs: list[Path], results_dir: Path, num_workers: int) -> None:
+    output = results_dir / "model-outputs-geojson"
 
-    GeoJSON files can be used with pathology viewers like QuPath.
-
-    RESULTS_DIR     Path to results directory (containing model-outputs dir).
-
-    OUTPUT          Path to output directory in which to save GeoJSON files.
-    """
     if not results_dir.exists():
-        raise click.ClickException(f"results_dir does not exist: {results_dir}")
-    if output.exists():
-        raise click.ClickException("Output directory already exists.")
+        raise FileExistsError(f"results_dir does not exist: {results_dir}")
     if (
-        not (results_dir / "model-outputs").exists()
+        not (results_dir / "model-outputs-csv").exists()
         and (results_dir / "patches").exists()
     ):
-        raise click.ClickException(
+        raise FileExistsError(
             "Model outputs have not been generated yet. Please run model inference."
         )
-    if not (results_dir / "model-outputs").exists():
-        raise click.ClickException(
-            "Expected results_dir to contain a 'model-outputs' directory but it does"
+    if not (results_dir / "model-outputs-csv").exists():
+        raise FileExistsError(
+            "Expected results_dir to contain a 'model-outputs-csv' directory but it does"
             " not. Please provide the path to the directory that contains"
             " model-outputs, masks, and patches."
         )
+    if output.exists():
+        geojsons = list((results_dir / "model-outputs-geojson").glob("*.json"))
 
-    csvs = list((results_dir / "model-outputs").glob("*.csv"))
-    if not csvs:
-        raise click.ClickException("No CSVs found. Did you generate model outputs?")
-
-    output.mkdir(exist_ok=False)
+        # Makes a list of filenames for both geojsons and csvs
+        geojson_filenames = [filename.stem for filename in geojsons]
```

Comment:
a nit.. instead of using `filenames` in the variable name, could you use `slide_ids` ? it will make it more obvious that these are slide IDs.